packages/dara-core/dara/core/internal/encoder_registry.py

```diff
@@ -114,5 +117,7 @@ def _df_decode_resolver(df: Any):
     pandas.Series: Encoder(serialize=lambda x: x.to_list(), deserialize=lambda x: pandas.Series(x)),
     pandas.Index: Encoder(serialize=lambda x: x.to_list(), deserialize=lambda x: pandas.Index(x)),
     pandas.Timestamp: Encoder(serialize=lambda x: x.isoformat(), deserialize=lambda x: pandas.Timestamp(x)),
-    pandas.DataFrame: Encoder(serialize=lambda x: x.to_json(orient='records'), deserialize=lambda x: _df_decode_resolver(x)),
+    pandas.DataFrame: Encoder(
+        serialize=lambda x: x.to_dict(orient='records'), deserialize=lambda x: _df_decode_resolver(x)
```

Comment:
Is orient='record' the one users would usually expect? cc @claudiawestby

Reply:
using records will eliminate the index which I would advise against. this is really just a matter of preference, at causalens we use `orient="list"` a lot because that's what `CausalNet` _used_ to expect but with 3.0 refactors it's no longer. `orient="list"` also removes the index so I wouldn't go with that one anyways

the default is `orient="dict"` which does keep the index so I would suggest sticking with that